clients/src/main/java/org/apache/kafka/clients/ApiVersions.java

```diff
@@ -30,18 +30,29 @@
  * NOTE: This class is intended for INTERNAL usage only within Kafka.
  */
 public class ApiVersions {
-
     private final Map<String, NodeApiVersions> nodeApiVersions = new HashMap<>();
     private byte maxUsableProduceMagic = RecordBatch.CURRENT_MAGIC_VALUE;
+    private ChangeListener changeListener = null;
```

Comment:
It's not really clear to me why we need this. We are using it to detect when the broker does not support the Metadata version we need for the auto creation flag, but it's not clear to me why we cannot wait until request time like usual. If we didn't have this, wouldn't we still fail in `MetadataRequest.Builder.build()` if `allowAutoTopicCreation` is enabled and the broker doesn't support it?